blocks/form/form.js

```diff
@@ -0,0 +1,118 @@
+import createField from './form-fields.js';
+import { div, form as Form, input } from '../../scripts/dom-helpers.js';
+import { getPathSegments, loadTranslations, translate } from '../../scripts/utils.js';
+
+async function createForm(formHref, submitHref) {
+  const resp = await fetch(formHref);
+  const json = await resp.json();
+  const [locale] = getPathSegments();
+  // get translations
+  await (async () => {
+    await loadTranslations(`${formHref}?sheet=translations`, locale || 'en');
+  })();
+
+  const translatedData = json.data.map((fd) => {
+    fd.Label = translate(fd.Label);
+    return fd;
+  });
+
+  const form = Form();
+  form.dataset.action = submitHref;
+
+  const localeField = input({ hidden: 'hidden', name: 'locale', value: locale || 'en' });
+  form.append(localeField);
+
+  const fields = await Promise.all(translatedData.map((fd) => createField(fd, form)));
+
+  fields.forEach((field) => {
+    if (field) {
+      form.append(field);
+    }
+  });
+
+  // group fields into fieldsets
+  const fieldsets = form.querySelectorAll('fieldset');
+  fieldsets.forEach((fieldset) => {
+    form.querySelectorAll(`[data-fieldset="${fieldset.name}"`).forEach((field) => {
+      fieldset.append(field);
+    });
+  });
+
+  return form;
+}
+
+function generatePayload(form) {
+  const payload = {};
+
+  [...form.elements].forEach((field) => {
+    if (field.name && field.type !== 'submit' && !field.disabled) {
+      if (field.type === 'radio') {
+        if (field.checked) payload[field.name] = field.value;
+      } else if (field.type === 'checkbox') {
+        if (field.checked) payload[field.name] = payload[field.name] ? `${payload[field.name]},${field.value}` : field.value;
+      } else {
+        payload[field.name] = field.value;
+      }
+    }
+  });
+  return payload;
```

Comment:
Should we add a random id/uuid to each form submission? Not sure what is the best practise in EDS around form submission to sheet, so adding this comment for a suggestion.